application/schemas/templates/components.py

```diff
@@ -28,6 +33,26 @@ class Component(BaseModel):
                     'a higher priority.',
     )
 
+    deploy: Optional[httpDeploy] = Field(description='Deployment details for HTTP type components')
+    status: Optional[httpStatus] = Field(description='Status details for HTTP type components')
+    delete: Optional[httpDelete] = Field(description='Deletion details for HTTP type components')
+
+
+    @root_validator(pre=True)
+    def check_deploy(cls, values):
+        type_ = values.get('type')
+        deploy = values.get('deploy')
+        delete = values.get('delete')
+        if type_ == ComponentTypes.http:
+            if deploy is None:
+                raise ValueError('deploy is required when type is http')
+            if delete is None:
+                raise ValueError('delete is required when type is http')
+        return values
+
+    def __hash__(self) -> str:
```

Comment:
Could you please clarify why did you implement this? I can't find where actually you need this.

Reply:
This was needed for the install operation.